stock_picking_partner_block_out/models/stock_picking.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2024 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class StockPicking(models.Model):
+
+    _inherit = "stock.picking"
+
+    partner_picking_block_out = fields.Boolean(
+        related="partner_id.picking_block_out", store=True
+    )
+
+    @api.constrains("partner_picking_block_out", "state")
```

Comment:
Can you check if you also have to add `picking_type_code` and `is_dropship` (who are behind `_is_to_external_location()`)

Reply:
I have modified my constraint because indeed, I should depend on `picking_type_code` but IMO, I shouldn't add 'is_dropship' since it is introduced by the module `stock_dropshipping`

Reply:
I changed my mind about adding `picking_type_code`:
* This field is a related -> I should store the field to be able to use add it in `@api.constrains`
* Changing the picking type is only possible in 'draft' state so in my case, it is not necessary to trigger the constraint when modifying the picking type